server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java

```diff
@@ -131,4 +139,100 @@ public void validateHealthchecksShouldReturnInternalErrorWhenOneHealthCheckIsDeg
     }
+    
+    @Test
+    public void performHealthCheckShouldReturnOkWhenHealthCheckIsHealthy() {
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_1)));
+        
+        given()
+            .pathParam("componentName", COMPONENT_NAME_1.getName())
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("componentName", equalTo("component-1"))
+            .body("escapedComponentName", equalTo("component-1"))
+            .body("status", equalTo(ResultStatus.HEALTHY.getValue()))
+            .body("cause", is(nullValue()));
+    }
+    
+    @Test
+    public void performHealthCheckShouldReturnNotFoundWhenComponentNameIsUnknown() {
+        
+        given()
+            .pathParam("componentName", "unknown")
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("details", is(nullValue()))
+            .body("type", equalTo("notFound"))
+            .body("message", equalTo("Component with name unknown cannot be found"))
+            .body("statusCode", is(404));
+    }
+    
+    @Test
+    public void performHealthCheckShouldReturnInternalErrorWhenHealthCheckIsDegraded() {
+        healthChecks.add(healthCheck(Result.degraded(COMPONENT_NAME_1, "the cause")));
+        
+        given()
+            .pathParam("componentName", COMPONENT_NAME_1.getName())
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+            .body("componentName", equalTo("component-1"))
+            .body("escapedComponentName", equalTo("component-1"))
+            .body("status", equalTo(ResultStatus.DEGRADED.getValue()))
+            .body("cause", equalTo("the cause"));
+    }
+    
+    @Test
+    public void performHealthCheckShouldReturnInternalErrorWhenHealthCheckIsUnhealthy() {
+        healthChecks.add(healthCheck(Result.unhealthy(COMPONENT_NAME_1)));
+        
+        given()
+            .pathParam("componentName", COMPONENT_NAME_1.getName())
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+            .body("componentName", equalTo("component-1"))
+            .body("escapedComponentName", equalTo("component-1"))
+            .body("status", equalTo(ResultStatus.UNHEALTHY.getValue()))
+            .body("cause", is(nullValue()));
+    }
+    
+    @Test
+    public void performHealthCheckShouldReturnProperlyEscapedComponentName() {
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_3)));
+        
+        given()
+            .pathParam("componentName", COMPONENT_NAME_3.getName())
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .body("componentName", equalTo("component 3"))
```

Comment:
I see we use `"component 3"` in many places, could you extract it into an constant? and `"component%203"` too.